### PR TITLE
Improve tenant domain resolving logic in recovery endpoint

### PR DESCRIPTION
**Pull request by emswbandara**
Resolves https://github.com/wso2/product-is/issues/8130

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp [hunk 1]
```diff
@@ -35,7 +36,12 @@
     if (StringUtils.isBlank(callback)) {
         callback = request.getParameter("redirect_uri");
     }
-    String tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+    String tenantDomain;
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+        tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+    } else {
+        tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+    }
```
Comment (malithie) [inline apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp:44]: If this logic is there in all jsps, shall we look into separating it to a common one and reuse. Check if such common jsp is already available

File: apps/recovery-portal/src/main/webapp/tenant-resolve.jsp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+<%
+    String tenantDomain;
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+        tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+    } else {
+        tenantDomain = request.getParameter("tenantDomain");
+        if (StringUtils.isBlank(tenantDomain)) {
+            tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+        }
```
Comment (malithie) [inline apps/recovery-portal/src/main/webapp/tenant-resolve.jsp:31]: Is this constant different from above ?

File: apps/recovery-portal/src/main/webapp/username-recovery-tenant-request.jsp [hunk 3]
```diff
@@ -71,9 +72,19 @@
...
+                    %>
                     <input id="tenant-domain" type="text" name="tenantDomain"
                                 class="form-control ">
-                        
+                    <%
+                        } else {
+                    %>
+                    <input id="tenant-domain" type="hidden" name="tenantDomain" value='<%=Encode.forHtmlAttribute
+                                (IdentityTenantUtil.getTenantDomainFromContext())%>' class="form-control">
+                    <%
```
Comment (malithie) [inline apps/recovery-portal/src/main/webapp/username-recovery-tenant-request.jsp:85]: Why tenant domain had to be passed as a parameter in this case

File: apps/recovery-portal/src/main/webapp/tenant-resolve.jsp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+%>
```
Comment (ashensw) [inline apps/recovery-portal/src/main/webapp/tenant-resolve.jsp:33]: Add a new line.

File: apps/recovery-portal/src/main/webapp/tenant-resolve.jsp (hunk 2 above)
Comment (emswbandara) [reply apps/recovery-portal/src/main/webapp/tenant-resolve.jsp:31]: yes , the value of this constant is "tenantdomain".

File: apps/recovery-portal/src/main/webapp/username-recovery-tenant-request.jsp (hunk 3 above)
Comment (emswbandara) [reply apps/recovery-portal/src/main/webapp/username-recovery-tenant-request.jsp]: fixed

File: apps/recovery-portal/src/main/webapp/tenant-resolve.jsp (hunk 4 above)
Comment (emswbandara) [reply apps/recovery-portal/src/main/webapp/tenant-resolve.jsp]: fixed

**Pull request merged**